### ISSUE TITLE
schematic only saving/loading in half

### ISSUE THREAD

**Issue by TobiasDeBruijn** [Lunatrius/Schematica#359]
**Bug:**
I made a schematic of a pretty big build, I dont know if it happens during the saving or loading. See pic
![2018-07-13_21 11 30](https://user-images.githubusercontent.com/28569170/42709601-589db548-86e1-11e8-917d-fec79a0633db.png)
![2018-07-13_21 11 37](https://user-images.githubusercontent.com/28569170/42709605-5b2749e6-86e1-11e8-82b5-0045b69406ae.png)

**Log:**
[fml-client-latest.log](https://github.com/Lunatrius/Schematica/files/2193900/fml-client-latest.log)
[latest.log](https://github.com/Lunatrius/Schematica/files/2193901/latest.log)

**Reproduce:**
big build > make schematic (format alpha) > load in schematic

**Version:** 
mc: 1.12.2 
forge: 1.12.2-14.23.2.2611
schematica: Schematica-1.12.2-1.8.0.168-universal
LunatriusCore: LunatriusCore-1.12.2-1.2.0.42-universal

**Comment by Lunatrius** [comment]
What are the coordinates at that place? Can you check if the printer puts down any blocks where there's supposedly no blocks?

**Comment by monteirodan** [comment]
It happens the same issue with me. I copied an wide area from my base on a multiplayer server to paste in a single player world. Then always render only the half. Then I tried to subdivide the area and copy by parts (what makes better to process with low memory). But still happens. Now I'll try to subdivide a little smaller. If I got I'll tell here.

MC version 1.12.2
Forge version 14.23.4.2705

**Comment by Lunatrius** [comment]
Same question
> What are the coordinates at that place? Can you check if the printer puts down any blocks where there's supposedly no blocks?

**Comment by Lunatrius** [comment]
Closing due to lack of feedback.

**Comment by BZ2600** [comment]
Yes, I am having the same issue.  Although I have done a bit of testing (Very little), and it does appear to be a limitation in Schematica.
-
Win7Ultimate 64, Schematica-1.12.1-1.8.0.164-universal.jar, LunatriusCore-1.12.1-1.2.0.41-universal.jar in Minecraft 1.12.1, forge-1.12.1-14.22.1.2478-installer.jar, Java 1.8.0_51 64bit.
-
Multiplayer: (Origination)
Red point: x:1003048,y:5,z:994409
Blue point: x:1003210,y:95,z:994571
Singleplayer: (Destination)
Red point: x:276,y:4,z:-10
Blue point: x:438,y:94,z:152
-
At the destination the entire outline of the schematic is there but the ghost image of the blocks is incomplete.  It starts at the Red point as it should then stops only showing the ghost blocks in the following cuboid . . .
Red point: x:276,y:4,z:-10
Blue point: x379,y:94,z:92
-
Printing blocks in Singleplayer Creative Mode only works where the Ghost Blocks are visible.  Apparently block type does not matter as various types of blocks were used (Printed/Not Printed).  All block types show as red when manually placed outside the displayed cuboid, but within the schematic area.  Within the displayed cuboid the schematic works as designed.
This has been tested reversing the Red and Blue points at the origination with the displayed cuboid always cornered on the Red point at the destination.
-
When at the origination world, saving the schematic output reads . . .
Started saving 121 chunks into name.schematic
Successfully saved name.schematic
When at the destination world with the reported 121 chunk schematic loaded, saving the displayed cuboid as a new schematic output reads . . .
Started saving 49 chunks into name.schematic
Successfully saved name.schematic
 . . . This discrepancy does not prove the issue is at origination or destination.

Thanks
BZ2600

**Comment by BZ2600** [comment]
Yes, I am having the same issue.  Although I have done a bit of testing (Very little), and it does appear to be a limitation in Schematica.
-
Win7Ultimate 64, Schematica-1.12.1-1.8.0.164-universal.jar, LunatriusCore-1.12.1-1.2.0.41-universal.jar in Minecraft 1.12.1, forge-1.12.1-14.22.1.2478-installer.jar, Java 1.8.0_51 64bit.
-
Multiplayer: (Origination)
Red point: x:1003048,y:5,z:994409
Blue point: x:1003210,y:95,z:994571
Singleplayer: (Destination)
Red point: x:276,y:4,z:-10
Blue point: x:438,y:94,z:152
-
At the destination the entire outline of the schematic is there but the ghost image of the blocks is incomplete.  It starts at the Red point as it should then stops only showing the ghost blocks in the following cuboid . . .
Red point: x:276,y:4,z:-10
Blue point: x379,y:94,z:92
-
Printing blocks in Singleplayer Creative Mode only works where the Ghost Blocks are visible.  Apparently block type does not matter as various types of blocks were used (Printed/Not Printed).  All block types show as red when manually placed outside the displayed cuboid, but within the schematic area.  Within the displayed cuboid the schematic works as designed.
This has been tested reversing the Red and Blue points at the origination with the displayed cuboid always cornered on the Red point at the destination.
-
When at the origination world, saving the schematic output reads . . .
Started saving 121 chunks into name.schematic
Successfully saved name.schematic
When at the destination world with the reported 121 chunk schematic loaded, saving the displayed cuboid as a new schematic output reads . . .
Started saving 49 chunks into name.schematic
Successfully saved name.schematic
 . . . This discrepancy does not prove the issue is at origination or destination.

**Comment by colocated** [comment]
Personally, I think schematica can only save whats in your render distance. So technically, if you can see all of your build at once, then it theoretically should be able to save it all.

**Comment by XilefTech** [comment]
> Personally, I think schematica can only save whats in your render distance. So technically, if you can see all of your build at once, then it theoretically should be able to save it all.

well, I just got that exact same problem as well, yesterday. While trying to figure out, why it wouldn't save properly, I noticed, it is related to your render distance, but not the only thing that it is dependent on. When I had selected it the first time, and saved it, the rendered chunks definitely went beyond what was saved. Also, it wasn't a straight line like what schematica exported. When trying to select bigger areas, I had success actually copying what I wanted to copy, yet it still wasn't my render distance... I had my render distance set to 33 and almost all chunks loaded in the space, that I selected way bigger than the object I wanted to save.
In the end, it barely saved what I wanted it to, nothing more, even though I clearly had a lot more chunks loaded...
Sadly I don't have the exact coordinates anymore.
